plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/flex/FlexGlobalDeployArtifactStagingDelegateTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.deploy.flex;
+
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.StagingDelegate;
+import java.io.File;
+import java.io.IOException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Path;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.osgi.service.prefs.BackingStoreException;
+
+public class FlexGlobalDeployArtifactStagingDelegateTest {
+
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  private final FlexGlobalDeployPreferences preferences = new FlexGlobalDeployPreferences();
+
+  private IPath stagingDirectory;
+  private IPath appEngineDirectory;
+  private File deployArtifact;
+
+  @Before
+  public void setUp() throws IOException, BackingStoreException {
+    stagingDirectory = new Path(tempFolder.getRoot().toString()).append("stagingDirectory");
+    appEngineDirectory = new Path(tempFolder.newFolder("appEngineDirectory").toString());
+    appEngineDirectory.append("app.yaml").toFile().createNewFile();
+    deployArtifact = tempFolder.newFile("my-app.war");
+    preferences.setDeployArtifactPath(deployArtifact.toString());
```

Comment:
you should use some combination of getPath/getAbsolutePath/getCanonicalPath/getName instead of toString here for clarity as to what's expected if nothing else

Reply:
Making every path absolute probably provides the best clarity.